### PR TITLE
Create tray with icon

### PR DESCRIPTION
**Pull request by opjs105rogue**
I relied on the example of the Telegram icon for functionality.
The feature of enabling the application by double-clicking on the tray icon has not yet been implemented

### CODE REVIEW

**Comment by agoalofalife** [comment]
CleanShot 2024-09-29 at 16 40 16
I think it doesn't work, png is not suitable for tray

Try to find some svg picture and possible to change size relatively bar

**Comment by opjs105rogue** [comment]
tried to fix it with resize method, It should work

File: main.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
-const { app, BrowserWindow } = require("electron");
```
Comment (agoalofalife) [inline main.js]: we have to change require to import everywhere

File: main.js [hunk 2]
```diff
@@ -17,10 +29,45 @@ app.whenReady().then(() => {
   app.on('activate', () => {
     if (BrowserWindow.getAllWindows().length === 0) createWindow()
   })
+
+
+  // Add icon for Tray bar
+  const path = require('path');
```
Comment (agoalofalife) [inline main.js]: all dependencies has to be on the top of this file

**Comment by opjs105rogue** [comment]
Changed the requie function to import and lifted them up the code. New variables have also been added to work with path

File: main.js [hunk 3]
```diff
@@ -1,14 +1,28 @@
-const { app, BrowserWindow } = require("electron");
+import { app, BrowserWindow, Tray, Menu, nativeImage } from "electron";
+import { fileURLToPath } from 'url';
+import path, { dirname } from 'path';
+
+let win;
```
Comment (agoalofalife) [inline main.js]: it's not good name for variable - and I think shortness is  inappropriate- win -> window

File: main.js [hunk 4]
```diff
@@ -1,14 +1,28 @@
...
+let tray;
```
Comment (agoalofalife) [inline main.js]: I don't see a reason, why this variable(tray) in above than function

File: main.js [hunk 5]
```diff
@@ -1,14 +1,28 @@
...
 
 const createWindow = () => {
```
Comment (agoalofalife) [inline main.js:9]: this function might return window and we can use it inside `app.whenReady().then(() => {` 
obviously don't need to save variable top of this file

**Comment by opjs105rogue** [comment]
HI, @agoalofalife , check please last commands: 
`// macOS apps generally continue running even without any windows open.
// Activating the app when no windows are available should open a new one.
app.on("activate", () => {
  if (BrowserWindow.getAllWindows().length === 0) createWindow();
});

//Quit the app when all windows are closed (Windows & Linux)
app.on("window-all-closed", () => {
  if (process.platform !== "darwin") app.quit();
});
`
Do they conflict with the logic of interaction with the tray icon? Because the general logic with closing \ hiding \ opening the application is already spelled out above, and I don't see the point in these lines.(at least when I delete them, nothing changes at all)

**Comment by agoalofalife** [comment]
@RGamz Why we need this lines, what they do?

**Comment by RGamz** [comment]
@agoalofalife  normally they should do what you see in comments accordnig to the electron docs.... but i never tried to delete them and see what happens )

**Comment by agoalofalife** [comment]
@opjs105rogue 
Long story short, I dive into this topic a little bit. First of all the initial icon does't appear in my tray bar. The problem is depend on size picture and some settings. I recommend to read this section https://www.electronjs.org/docs/latest/api/native-image#template-image-macos
Also , pay attention for this section https://www.electronjs.org/docs/latest/api/native-image#template-image-macos

Obviously we have to get smaller picture and play by settings, Do you have mac os?

**Comment by opjs105rogue** [comment]
@agoalofalife , Hi! Сheck this and test it on your system, please, unfortunately, I couldn't run macOS on a virtual machine, so I have to work with what I have.
Changed the images, reduced the size and added suffixes for DPI + add Template.
It is possible to prescribe the logic of the tray icon for each system separately, but for now let's focus on whether the icon is working correctly on macOS now?
do not pay attention to the "nativeTheme" module, I will delete it or continue working with it according to the logic of interaction with the OC after your response(I forgot to delete it right away)

**Comment by agoalofalife** [comment]
https://github.com/user-attachments/assets/35e71540-2303-4b5a-b9e6-e54a025a6730

I am demonstrating icon on the video. Also, I managed to fix click by icon.
Now single click open and hide main window. Right-click show context menu.
Reference https://github.com/electron/electron/issues/2830
FYI @RGamz

**Comment by opjs105rogue** [comment]
Hi, @agoalofalife , as I found out, the Linux system's problem with processing clicks and interacting with the tray icon is quite old, but it has not been fixed in electron. 
https://stackoverflow.com/questions/57209717/electron-trayicon-not-working-properly-on-linux
https://github.com/electron/electron/issues/14941
Please look at my solution to this problem, since Linux handles clicks in a peculiar way, we will give you the opportunity to open the context menu with both the right and left mouse buttons on LinuxOC. At least for me it works correctly and cool on Linux

**Review by agoalofalife — APPROVED** [review]
Good Job!

**Pull request merged**